src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -379,6 +379,9 @@ public void Reset()
             _requestHeadersParsed = 0;
 
             _responseBytesWritten = 0;
+
+            // When testing parser can be null
+            _parser?.Reset();
```

Comment:
Why not automatically reset when done with headers? Dispose I get, but this? YAGI.